kubernetes/nodecollection.go

```diff
@@ -391,3 +394,18 @@ func buildContainersRequest() ([]byte, error) {
 	}
 	return body, nil
 }
+
+// GetNodeCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
```

Comment:
Should this be `Returns -1 and nil`?

Also, would it make sense to have this method return a Go error rather than the index - since, besides the potential debug log, we only check the index it >= 0.

Reply:
Since this is based on https://github.com/kubernetes/kubernetes/blob/v1.17.3/pkg/controller/util/node/controller_utils.go#L286 I'd like to keep it as close as possible.